### PR TITLE
feat(have-i-been-pwned): init

### PR DESCRIPTION
**Pull request by trinkey**
## 🎉 Theme for Have I Been Pwned 🎉
A website for checking if your emails/passwords have been exposed in data breaches

## 💬 Additional Comments 💬
no

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

### CODE REVIEW

File: styles/haveibeenpwned/catppuccin.user.css [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+      & when (@lookup = frappe) {
+        filter: brightness(0) saturate(100%) invert(84%) sepia(13%) saturate(1084%) hue-rotate(193deg) brightness(99%) contrast(94%);
+      }
+      & when (@lookup = macchiato) {
+        filter: brightness(0) saturate(100%) invert(86%) sepia(6%) saturate(1660%) hue-rotate(195deg) brightness(99%) contrast(94%);
+      }
+      & when (@lookup = mocha) {
+        filter: brightness(0) saturate(100%) invert(81%) sepia(8%) saturate(766%) hue-rotate(191deg) brightness(106%) contrast(91%);
+      }
+    }
```
Comment (uncenter) [inline styles/haveibeenpwned/catppuccin.user.css:105]: This is fine, just make sure you are getting results with a very precision level (<0.3 on https://angel-rs.github.io/css-color-filter-generator/). I [wrote this script](https://discord.com/channels/907385605422448742/1030437391607025765/1258863184723316797) one time that you can paste in the console to keep computing results until it is within range:

```js
while (Number.parseFloat(document.querySelector('#lossDetail').firstChild.textContent.split(':')[1].trim().slice(0, -1)) > 0.3) {
  compute()
}
```

**Review by uncenter — CHANGES_REQUESTED** [review]
Only this one comment on initial review (haven't tested the website appearance yet though)! Really great job, thanks for adding this!

File: styles/haveibeenpwned/catppuccin.user.css (hunk 1 above)
Comment (trinkey) [reply styles/haveibeenpwned/catppuccin.user.css:105]: I'm pretty sure I did, although I can re-compute to make sure I get below that threshold

Comment (uncenter) [reply styles/haveibeenpwned/catppuccin.user.css]: Thank you! I understand it is a pain in the ~~ass~ to do that. We are working (sorta) on adding documentation for the filter trick, and we will provide a list of the filters pre-computed in the future.

File: styles/haveibeenpwned/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+      & when (@lookup = frappe) {
+        filter: brightness(0) saturate(100%) invert(74%) sepia(16%) saturate(454%) hue-rotate(192deg) brightness(110%) contrast(92%);
+      }
+      & when (@lookup = macchiato) {
+        filter: brightness(0) saturate(100%) invert(82%) sepia(77%) saturate(662%) hue-rotate(181deg) brightness(103%) contrast(92%);
+      }
+      & when (@lookup = mocha) {
+        filter: brightness(0) saturate(100%) invert(88%) sepia(10%) saturate(1247%) hue-rotate(191deg) brightness(99%) contrast(93%);
+      }
+    }
```
Comment (somerand0mcat) [inline styles/haveibeenpwned/catppuccin.user.css:105]: can you use variables for filters like [this](https://github.com/search?q=repo:catppuccin/userstyles+%22filter:+@%22&type=code)?

**Review by somerand0mcat — CHANGES_REQUESTED** [review]

**Comment by somerand0mcat** [comment]
nice work! this is probably going to be the 26th init PR which is under 5 comments lol

the only unthemed stuff i've found is these unthemed icons/buttons
image

**Comment by trinkey** [comment]
> the only unthemed stuff i've found is these untamed icons/buttons

i can fix this at ~7:30 est (6 hours from now) but i gotta head to work right now

**Comment by trinkey** [comment]
> i can fix this at ~7:30 est (6 hours from now) but i gotta head to work right now

not me lying

anyways today was a really really *really* busy day ill do it tomorrow

**Review by isabelroses — DISMISSED** [review]

File: scripts/userstyles.yml [hunk 3]
```diff
@@ -416,6 +416,13 @@ userstyles:
     readme:
       app-link: "https://news.ycombinator.com"
     current-maintainers: [*lucasmelin]
+  haveibeenpwned:
```
Comment (uncenter) [inline scripts/userstyles.yml:419]: Probably should do proper kebab case right?
```suggestion
  have-i-been-pwned:
```

Comment (trinkey) [reply scripts/userstyles.yml:419]: i suppose

Comment (uncenter) [reply scripts/userstyles.yml:419]: Make sure to update the folder name, userstyles.yml metadata, and usercss metadata.

**Review by uncenter — APPROVED** [review]

**Review by isabelroses — APPROVED** [review]

**Review by uncenter — CHANGES_REQUESTED** [review]
Just noticed that if you click "pwned?" button next to the search bar with nothing in the input field, the resulting `#invalidAccount` container/banner is an unthemed shade of yellow.

**Comment by sgoudham** [comment]
The only thing I noticed is that the icons are hard to look at on Latte:

image

Unsure if this is something you can fix but figured I'd report it anyways, nice job on the theme so far 👍

**Comment by uncenter** [comment]
There are about a million of those icons for every company/breach, but maybe theming the two default icons (lined paper / list and mail / letter) could be worth it. They are PNGs so you'd have to use the filter technique, which I see you are familiar with for the logo (this time you can put the filters in the palette down below though as to not repeat yourself, see: https://github.com/catppuccin/userstyles/blob/a1f3fefcbf7ccc97bc4567e26c54f154a14f9566/styles/deepl/catppuccin.user.css#L326-L332)

**Comment by trinkey** [comment]
> The only thing I noticed is that the icons are hard to look at on Latte:

I could go through and add more filters for more icons but i don't know if I want to spend 3 hours adding filters to pngs, its probably fine enough as it is at the moment

**Comment by thismoon** [comment]
i noticed a very small thing. the `.unverified` class is unthemed (colored #a65c5c). for example the check for `example@example.org` has a first result called "Separate Data Breaches" that has a `span.unverified` element with unthemed parentheses (actually text inside the parentheses should be dark red/alternate color too)
no theme
image
catppuccin
image
suggested change:
```less
.unverified, .unverified a{
    color: darken(@red, 50%) !important
}
```
image

**Review by sgoudham — COMMENTED** [review]
Looks good to me, will leave it for @uncenter's final approval.

**Review by uncenter — APPROVED** [review]

**Review by isabelroses — APPROVED** [review]

**Pull request merged**